### ISSUE TITLE
Channel.non_blocking_select not working with when

### ISSUE THREAD

**Issue by j8r** [crystal-lang/crystal#8409]
By looking how to fix CI warnings related to `select` being deprecated, I compiled the compiler against master.
Using `non_blocking_select` produces a compiler error: 

```cr
ch = Channel(Nil).new
non_blocking_select
# Or Channel.non_blocking_select
when m = ch.receive
end
```

```
 3 | when m = ch.receive
     ^
Error: expecting token 'EOF', not 'when'
```

**Comment by j8r** [comment]
Related commit: https://github.com/crystal-lang/crystal/commit/1254fd2fc87c5ea65f68bf06d9e75d5a49ee95ce

**Comment by asterite** [comment]
Check this comment:

https://github.com/crystal-lang/crystal/commit/1254fd2fc87c5ea65f68bf06d9e75d5a49ee95ce#diff-28534fbfedc125e25dfc58732fa05c52R303-R307

The compiler will eventually produce calls to `Channel.select` or `Channel.non_blocking_select`

It's not that we'll introduce a `non_blocking_select` keyword. I don't think there's anything to do here, @bcardiff or @waj will later tackle this.

**Comment by j8r** [comment]
So it's still WIP, we can't do anything to remove this warnings?

**Comment by asterite** [comment]
Wait for the core team to handle that TODO (well, it should have been a TODO).

**Comment by bcardiff** [comment]
The warning is only produced by a 0.31 compiler against master.
Using a 0.32-dev compiler is already warning free related to this deprecation.

The only thing left is to remove the deprecation after 0.32. I don't see a need for a `# TODO`

**Comment by asterite** [comment]
Oh, sorry, I think I misunderstood this.

However, there's still an existing call that produces the warning:

https://github.com/crystal-lang/crystal/blob/f9a4a14a7f92c15f19086e983b18645b2738727e/spec/std/channel_spec.cr#L365

Can that be fixed now or we need to wait for 0.32.0? Maybe that's the warning that appears in CI.

**Comment by bcardiff** [comment]
Well, deprecated method should still work. So there should be specs. (Otherwise we can fall in https://github.com/crystal-lang/crystal/pull/8257#discussion_r333728144)

The behaviour of `non_blocking_select` is tested in  `describe ".non_blocking_select"` so that spec will go away.

This push the story of _allow warnings in within these lines_ :-)
